### PR TITLE
Feat/component

### PR DESCRIPTION
**Pull request by jgfidelis**
https://fidelis--delivery.myvtex.com/pizza-pepperoni/p

### CODE REVIEW

File: react/index.tsx [hunk 1]
```diff
@@ -0,0 +1,88 @@
+import React, { Fragment, FC, useRef, useEffect } from 'react'
+import { generateBlockClass, BlockClass } from '@vtex/css-handles'
+//@ts-ignore
+import styles from './styles.css'
+import { path } from 'ramda'
+
+enum POSITIONS {
```
Comment (lbebber) [inline react/index.tsx]: I think it's more standard for enums to have CapitalizedCamelCase

File: react/index.tsx [hunk 2]
```diff
@@ -0,0 +1,88 @@
+import React, { Fragment, FC, useRef, useEffect } from 'react'
+import { generateBlockClass, BlockClass } from '@vtex/css-handles'
+//@ts-ignore
+import styles from './styles.css'
```
Comment (lbebber) [inline react/index.tsx]: nit, but move styles to the bottom of the imports in a separate group

File: react/index.tsx [hunk 3]
```diff
@@ -0,0 +1,88 @@
...
+import { path } from 'ramda'
+
+enum POSITIONS {
+  BOTTOM = 'BOTTOM',
```
Comment (lbebber) [inline react/index.tsx]: also the actual value of the enum should be lower case I think

File: react/index.tsx [hunk 4]
```diff
@@ -0,0 +1,88 @@
...
+    }
+    const newTop =
+      window.innerHeight -
+      startTop.current -
+      container.current.clientHeight +
+      window.pageYOffset
+
+    if (newTop < 0) {
+      hasTouched.current = true
+      container.current.style.transform = `translateY(${newTop}px)`
```
Comment (lbebber) [inline react/index.tsx]: it's a hack, but `translate3d(x,y,z)` has generally better performance (bc it forces the element to be in a separate composition layer)

**Review by lbebber — CHANGES_REQUESTED** [review]

File: README.md [hunk 5]
```diff
@@ -1 +1,53 @@
...
+
+`PositionEnum` description:
+| Enum name | Enum value | Description |
+| --------- | ---- | ----------- |
+| BOTTOM | 'BOTTOM' | Component will stick to the bottom of screen |
+
+### Example usage
+
+```
+"store.product": {
```
Comment (klzns) [inline README.md]:
```suggestion
  "store.product": {
```

File: README.md [hunk 6]
```diff
@@ -1 +1,53 @@
...
+| `position`   | `PositionEnum` | Indicates where the component should stick                | N/A           |
+
+`PositionEnum` description:
+| Enum name | Enum value | Description |
+| --------- | ---- | ----------- |
+| BOTTOM | 'BOTTOM' | Component will stick to the bottom of screen |
+
+### Example usage
+
+```
```
Comment (klzns) [inline README.md]:
```suggestion
```js
```

File: README.md [hunk 7]
```diff
@@ -1 +1,53 @@
...
+  },
+  "flex-layout.row#buy-button": {
+    "props": {
+      "marginTop": 4,
+      "marginBottom": 7,
+      "paddingBottom": 2
+    },
+    "children": ["buy-button"]
+  },
+}
```
Comment (klzns) [inline README.md]:
```suggestion

```

File: manifest.json [hunk 8]
```diff
@@ -0,0 +1,16 @@
+{
+  "vendor": "vtex",
+  "name": "sticky-layout",
+  "version": "0.0.1",
+  "title": "Modal",
```
Comment (klzns) [inline manifest.json]:
```suggestion
  "title": "Sticky Layout",
```

File: react/.eslintrc [hunk 9]
```diff
@@ -0,0 +1,19 @@
+{
+  "extends": "vtex",
```
Comment (klzns) [inline react/.eslintrc]: Setup the right eslint
https://docs.google.com/document/d/1ePi5Mn3pe1dOorKrMtC-MclBDiIDimVzf0Uj61W8lA4/edit

File: react/index.tsx [hunk 10]
```diff
@@ -0,0 +1,88 @@
+import React, { Fragment, FC, useRef, useEffect } from 'react'
+import { generateBlockClass, BlockClass } from '@vtex/css-handles'
+//@ts-ignore
```
Comment (klzns) [inline react/index.tsx]: Add definition for css files 
https://github.com/vtex-apps/base-store-component/blob/master/react/typings/css.d.ts

File: react/index.tsx [hunk 11]
```diff
@@ -0,0 +1,88 @@
...
+  const container = useRef<HTMLDivElement>(null)
+  const startTop = useRef<number | null>(null)
+  const hasTouched = useRef<boolean>(false)
+
+  if (position !== POSITIONS.BOTTOM) {
+    // Only 'BOTTOM' position supported for now!
+    return null
+  }
+
+  const handleScroll = () => {
```
Comment (klzns) [inline react/index.tsx]: This function should be inside the `useEffect`

File: react/index.tsx [hunk 12]
```diff
@@ -0,0 +1,88 @@
...
+  const viewOffset = path<number>(['current', 'offsetTop'], container)
+  useEffect(() => {
+    if (viewOffset != null && !hasTouched.current) {
+      startTop.current = viewOffset
+      handleScroll()
+    }
+  }, [viewOffset])
+
+  return (
+    <Fragment>
```
Comment (klzns) [inline react/index.tsx]: Remove Fragment

File: react/tsconfig.json [hunk 13]
```diff
@@ -0,0 +1,29 @@
...
+    "jsx": "react",
+    "lib": ["es2017", "dom", "es2018.promise"],
+    "module": "es6",
+    "moduleResolution": "node",
+    "noEmitOnError": true,
+    "noImplicitAny": true,
+    "noImplicitReturns": true,
+    "noImplicitThis": true,
+    "noUnusedLocals": true,
+    "noUnusedParameters": true,
```
Comment (klzns) [inline react/tsconfig.json:13]: Check if it's right 
https://docs.google.com/document/d/1ePi5Mn3pe1dOorKrMtC-MclBDiIDimVzf0Uj61W8lA4/edit#heading=h.8db4tllwjh7n

**Review by klzns — CHANGES_REQUESTED** [review]

File: README.md (hunk 6 above)
Comment (jgfidelis) [reply README.md]: its a json :P

File: README.md [hunk 14]
```diff
@@ -1 +1,52 @@
 # VTEX Sticky Layout
+
+This component is used to make its children have sticky behaviour.
+
+For now, it supports only one postion, with value "BOTTOM".
+
+Passing the prop `"position": "BOTTOM"` in your blocks.json, makes its children stick to the bottom of its view, while it should be hidden and once it reaches the position it should be, it stops and stays there.
```
Comment (lbebber) [inline README.md]:
```suggestion
Passing the prop `"position": "bottom"` in your blocks.json, makes its children stick to the bottom of its view, while it should be hidden and once it reaches the position it should be, it stops and stays there.
```

**Review by klzns — APPROVED** [review]

**Pull request merged**